pkg/api/ref.go

```diff
@@ -58,16 +59,18 @@ func GetReference(obj runtime.Object) (*ObjectReference, error) {
 	}
 	version := ""
 	parsedSelfLink := versionFromSelfLink.FindStringSubmatch(meta.SelfLink())
-	if len(parsedSelfLink) < 2 {
-		if ForTesting_ReferencesAllowBlankSelfLinks {
-			version = "testing"
-		} else if meta.SelfLink() == "" {
-			return nil, ErrNoSelfLink
+	if mustValidateSelfLink(obj) {
```

Comment:
If you need to validate links, create a version of GetReference that accepts a boolean that forces to validate the link.